x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -831,7 +892,6 @@ protected void doSaveState(IndexerState indexerState, DataFrameIndexerPosition p
                             r -> {
                                 // for auto stop shutdown the task
                                 if (state.getTaskState().equals(DataFrameTransformTaskState.STOPPED)) {
-                                    onStop();
```

Comment:
Do we not need this?? I know all it does it log & audit a message

Reply:
@davidkyle `onStop` is called in the Async indexer when transitioning from `STOPPING -> STOPPED` and we call it directly if `stop` transitions the indexer directly to `STOPPED`. This resulted in there always being two log entries for stopping which confused me at first (thinking somebody called stop twice).